File: MyHordesOptimizerWebsite/src/app/_abstract_model/dto/expedition-part.dto.ts
import { CitizenExpeditionDTO } from './citizen-expedition.dto';
import { ExpeditionOrderDTO } from './expedition-order.dto';

export interface ExpeditionPartDTO {
    id?: number;
    expeditionId?: number;
    orders: ExpeditionOrderDTO[];
    citizens: CitizenExpeditionDTO[];
    position: number;
    path: string;
    direction?: string;
}

export interface ExpeditionPartShortDTO {
    id?: number;
    expeditionId?: number;
    ordersId: number[];
    citizensId: number[];
    position: number;
    path: string;
    direction?: string;
}

File: MyHordesOptimizerApi/MyHordesOptimizerApi/Dtos/MyHordesOptimizer/Expeditions/ExpeditionPartDto.cs
using System.Collections.Generic;

namespace MyHordesOptimizerApi.Dtos.MyHordesOptimizer.Expeditions
{
    public class ExpeditionPartDto
    {
        public int? Id { get; set; }
        public List<ExpeditionOrderDto>? Orders { get; set; }
        public List<ExpeditionCitizenDto>? Citizens { get; set; }
        public string? Path { get; set; }
        public string? Label { get; set; }
        public string? Direction { get; set; }
    }
}
